torch/_inductor/cudagraph_trees.py

```diff
@@ -1869,6 +1916,40 @@ def try_end_curr_warmup(self):
             self.current_node = None
             return
 
+        self.check_warn_on_unable_to_end_warmup(function_id)
+
+    def check_warn_on_unable_to_end_warmup(self, function_id: FunctionID):
+        # are we in a potential loop where we are unable to end warmup
+        if not (
+            function_id in self.warmed_up_functions
+            and function_id not in self.warned_functions
+            and self.in_new_torch_compile_invocation()
+        ):
+            return
+
+        prior_warmup = next(
+            (
+                node
+                for node in self.current_node._path_from_root
+                if node.wrapped_function.id == function_id
+            ),
+            None,
+        )
+
+        if prior_warmup is None:
+            return
+
+        # first node executed or same pattern
+        if prior_warmup.parent is None or (
+            prior_warmup.parent.wrapped_function == self.current_node.wrapped_function
+        ):
+            self.warned_functions.add(function_id)
+            warnings.warn(
+                "Unable to end warm up phase of CUDAGraphs because of pending, uninvoked backwards. "
+                "Consider running with torch.no_grad() or using torch._inductor.cudagraph_mark_step_begin() "
+                "before each model invocation"
```

Comment:
I feel like this can be worded more clearly

Reply:
Imagine the user hasn't read the manual. What is "warm up phase"? What exactly is the user supposed to do? This might be a good place to link to a wiki page which we can more easily evolve over time.

Reply:
Rephrased, will link to doc when it lands